packages/subcommands/src/lib/Subcommand.ts

```diff
@@ -0,0 +1,211 @@
+import {
+	fromAsync,
+	isErr,
+	type Args,
+	Command,
+	type PieceContext,
+	type ChatInputCommand,
+	UserError,
+	Identifiers,
+	type MessageCommand
+} from '@sapphire/framework';
+import type { Message } from 'discord.js';
+import {
+	ChatInputSubcommandMappings,
+	ChatInputSubcommandGroupMappings,
+	ChatInputSubcommandMappingValue,
+	ChatInputSubcommandToProperty,
+	MessageSubcommandMappings,
+	MessageSubcommandMappingValue,
+	MessageSubcommandToProperty,
+	type SubcommandMappingsArray,
+	MessageSubcommandGroupMappings
+} from './SubcommandMappings';
+import { type ChatInputSubcommandAcceptedPayload, Events, type MessageSubcommandAcceptedPayload } from './types/Events';
+
+export class SubCommandPluginCommand extends Command<Args, SubCommandPluginCommand.Options> {
```

Comment:
Shouldn't this be like in framework (relating to the generic for Args and options, so we can extend this class and specify different types for options and args)